src/bug-filing/azure-boards/azure-boards-settings-form.tsx

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Dropdown, IDropdownOption, TextField } from 'office-ui-fabric-react';
+import * as React from 'react';
+
+import { NamedSFC } from '../../common/react/named-sfc';
+import { SettingsFormProps } from '../types/settings-form-props';
+import {
+    AzureBoardsBugFilingService,
+    AzureBoardsBugFilingSettings,
+    AzureBoardsIssueDetailField,
+    AzureBoardsIssueDetailLocationDropdownOption,
+} from './azure-boards-bug-filing-service';
+
+export const AzureBoardsSettingsForm = NamedSFC<SettingsFormProps<AzureBoardsBugFilingSettings>>('AzureDevOpsSettingsForm', props => {
+    const options: AzureBoardsIssueDetailLocationDropdownOption[] = [
+        {
+            key: 'reproSteps',
+            text: 'Repro steps (default)',
+        },
+        {
+            key: 'description',
+            text: 'Description',
+        },
+    ];
+
+    const onProjectURLChange = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
+        const propertyName: keyof AzureBoardsBugFilingSettings = 'projectURL';
+        props.deps.userConfigMessageCreator.setBugServiceProperty(AzureBoardsBugFilingService.key, propertyName, newValue);
+    };
+
+    const onIssueDetailLocationChange = (event: React.FormEvent<HTMLElement>, newValue?: IDropdownOption) => {
```

Comment:
Can we declare the second parameter as `AzureBoardsIssueDetailLocationDropdownOption` instead of `IDropdownOption`?

Reply:
I guess you're concern is related to the cast `newValue.key as AzureBoardsIssueDetailField` a couple of lines below. That I understand but the signature of this function is tide to the `Dropdown` component `onChange` definition, which does require the `newValue` to be of type `IDropdownOption`.